theano/sandbox/scan_module/scan_op.py

```diff
@@ -342,10 +342,10 @@ def rval(p=p, i=node_input_storage, o=node_output_storage, n=node):
         return rval
 
     def grad(self, args, g_outs):
-        pass
+        raise NotImplementedError()
 
     def R_op(self, inputs, eval_points):
-        pass
+        raise NotImplementedError()
```

Comment:
I don't think `raise NotImplementedError()` is in any way useful for people looking to implement grad and/or R_op

The R_op() in PureOp does the right thing to signal that the method is not defined (which is not to raise NotImplementedError).  There is no "default implementation" for these methods.